drivers/scheduler/k8s/k8s.go

```diff
@@ -944,6 +948,65 @@ func (k *k8s) Describe(ctx *scheduler.Context) (string, error) {
 	return buf.String(), nil
 }
 
+func (k *k8s) ScaleApplication(ctx *scheduler.Context, newScaleFactorMap map[string]int32) error {
+	k8sOps := k8s_ops.Instance()
+	for _, spec := range ctx.App.SpecList {
+		logrus.Infof("Scale all Deployments")
+		if obj, ok := spec.(*apps_api.Deployment); ok {
+			dep, err := k8sOps.GetDeployment(obj.Name, obj.Namespace)
+			if err != nil {
+				return err
+			}
+			newScaleFactor := newScaleFactorMap[obj.Name+DeploymentSuffix]
+			*dep.Spec.Replicas = newScaleFactor
+			if _, err := k8sOps.UpdateDeployment(dep); err != nil {
+				return &scheduler.ErrFailedToUpdateApp{
+					App:   ctx.App,
+					Cause: fmt.Sprintf("Failed to update Deployment: %v. Err: %v", obj.Name, err),
+				}
+			}
+			logrus.Infof("Deployment %s scaled to %d successfully.", obj.Name, newScaleFactor)
+		} else if obj, ok := spec.(*apps_api.StatefulSet); ok {
+			logrus.Infof("Scale all Stateful sets")
+			dep, err := k8sOps.GetStatefulSet(obj.Name, obj.Namespace)
+			if err != nil {
+				return err
+			}
+			newScaleFactor := newScaleFactorMap[obj.Name+StatefulSetSuffix]
+			*dep.Spec.Replicas = newScaleFactor
+			if _, err := k8sOps.UpdateStatefulSet(dep); err != nil {
+				return fmt.Errorf("%v", &scheduler.ErrFailedToUpdateApp{
```

Comment:
change the error here to use ErrFailedToUpdateApp

Reply:
Fixed.